### PR TITLE
Bump Go and rancher/shepherd version and update code

### PR DESCRIPTION
**Pull request by valaparthvi**
### What does this PR do?
1. Bump Go to 1.21
2. Bump rancher/shepherd to https://github.com/rancher/shepherd/commit/3271d2ea92d5c270d81c5ba7be8e124dee443695
3. Fix the code due to breaking changes to r/s.
4. Parallelize support matrix tests for GKE and AKS.
5. Add `--wait` and `--force` flags to `eksctl delete cluster` command to ensure the cluster is deleted properly.
6. Modify `AddNodeGroup` function for all the operators to only use the CATTLE_TEST_CONFIG nodepool/group data as a template and not iterate over it.
7. Log cluster name along with the k8s version to easily identify tests they belong it.
8. Reposition functions in gke/helper_cluster.go
9. Only delete a cluster if `cluster!=nil` to avoid panics in AfterEach on a failed cluster creation in BeforeEach.

### Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):
Fixes part of #103

### Checklist:
- [x] Squashed commits into logical changes
- [ ] Documentation
- [x] GitHub Actions (if applicable)
**GKE:**
1. Support Matrix (nightly): https://github.com/rancher/hosted-providers-e2e/actions/runs/9362008149
2. P0 (nightly): https://github.com/rancher/hosted-providers-e2e/actions/runs/9361700783
3. P1 (nightly): https://github.com/rancher/hosted-providers-e2e/actions/runs/9378544815/job/25821978339, https://github.com/rancher/hosted-providers-e2e/actions/runs/9378544815, https://github.com/rancher/hosted-providers-e2e/actions/runs/9316874267
10. K8s Chart Support: https://github.com/rancher/hosted-providers-e2e/actions/runs/9353976032

**EKS:** https://github.com/rancher/hosted-providers-e2e/actions/runs/9369177373, https://github.com/rancher/hosted-providers-e2e/actions/runs/9380957101/job/25829200325, https://github.com/rancher/hosted-providers-e2e/actions/runs/9378945261/job/25823036461 (support matrix)
**AKS:** https://github.com/rancher/hosted-providers-e2e/actions/runs/9362636809, https://github.com/rancher/hosted-providers-e2e/actions/runs/9369181929/job/25793096360
### Special notes for your reviewer:
Performing Version Bump:
1. Replace contents of go.mod with the following
```go.mod
module github.com/rancher/hosted-providers-e2e

go 1.21
require (
    github.com/onsi/ginkgo/v2 latest
    github.com/onsi/gomega latest
    github.com/epinio/epinio latest
    github.com/rancher/shepherd 3271d2ea92d5         // rancher/shepherd HEAD commit
    github.com/stretchr/testify latest
    k8s.io/api v0.28.6
    k8s.io/apimachinery v0.28.6
)

replace (
    k8s.io/api => k8s.io/api v0.28.6
    k8s.io/client-go => github.com/rancher/client-go v1.28.6-rancher1
)
```
11. `go mod tidy` - this resolves all dependencies, missing ones too
12. `go mod vendor`

### CODE REVIEW

**Comment by cpinjani** [comment]
Please check failing CI actions

**Comment by valaparthvi** [comment]
@cpinjani the tests have been a bit flaky for GKE and EKS. GKE tests have been passing locally, EKS kinda hits different issues with resource limits, but I don't think that is strictly related to the changes made in this PR. I think the PR is RFR.

**Comment by valaparthvi** [comment]
> I noticed panic in gke-operator logs in the `UpdateNodePoolAutoscaling` functions, I need to investigate more.

This is panic is noticed when updating autoscaling for a GKE created via CLI (rancherv2.8-c0f2d83a748db73a9737102a89e17a4b162edda7-head). It is not related to this PR.

File: Makefile [hunk 1]
```diff
@@ -80,12 +80,23 @@ e2e-p1-provisioning-tests: deps ## Run the 'P1Provisioning' test suite for a giv
 	ginkgo ${STANDARD_TEST_OPTIONS} --focus "P1Provisioning" ./hosted/${PROVIDER}/p1/
 
 
-#TODO: Parallelize supportmatrix tests
+# Support Matrix test has not been parallelized for EKS because we hit resource limits when running it in parallel
 e2e-support-matrix-importing-tests: deps ## Run the 'SupportMatrixImporting' test suite for a given ${PROVIDER}
+ifeq (${PROVIDER}, eks)
 	ginkgo ${STANDARD_TEST_OPTIONS} --focus "SupportMatrixImporting" ./hosted/${PROVIDER}/support_matrix/
+else
+	ginkgo ${STANDARD_TEST_OPTIONS} --nodes 2 --focus "SupportMatrixImporting" ./hosted/${PROVIDER}/support_matrix/
```
Comment (cpinjani) [inline Makefile:88]: We can also remove --nodes 2, perhaps later.

**Review by cpinjani — APPROVED** [review]
lgtm

**Pull request merged**